File: exercise/java/day01/src/test/java/SantaCommunicatorTest.java
import communication.SantaCommunicator;
import doubles.TestLogger;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class SantaCommunicatorTest {
    private static final String DASHER = "Dasher";
    private static final String NORTH_POLE = "North Pole";
    private final int numberOfDaysToRest = 2;
    private final int numberOfDayBeforeChristmas = 24;
    private final TestLogger logger = new TestLogger();
    private SantaCommunicator communicator;

    @BeforeEach
    void setup() {
        this.communicator = new SantaCommunicator(numberOfDaysToRest);
    }

    @Test
    void composeMessage() {
        var message = communicator.composeMessage(DASHER, NORTH_POLE, 5, numberOfDayBeforeChristmas);
        assertThat(message).isEqualTo("Dear Dasher, please return from North Pole in 17 day(s) to be ready and rest before Christmas.");
    }

    @Test
    void shouldDetectOverdueReindeer() {
        var overdue = communicator.isOverdue(
                DASHER,
                NORTH_POLE,
                numberOfDayBeforeChristmas,
                numberOfDayBeforeChristmas,
                logger);

        assertThat(overdue).isTrue();
        assertThat(logger.getLog())
                .isEqualTo("Overdue for Dasher located North Pole.");
    }

    @Test
    void shouldReturnFalseWhenNoOverdue() {
        assertThat(
                communicator.isOverdue(
                        DASHER,
                        NORTH_POLE,
                        numberOfDayBeforeChristmas - numberOfDaysToRest - 1,
                        numberOfDayBeforeChristmas,
                        logger)
        ).isFalse();
    }
}

File: exercise/kotlin/day01/src/test/kotlin/SantaCommunicatorTest.kt
package communication

import doubles.TestLogger
import io.kotest.core.spec.style.DescribeSpec
import io.kotest.matchers.shouldBe

class SantaCommunicatorTest : DescribeSpec({
    val numberOfDaysToRest = 2
    val numberOfDayBeforeChristmas = 24
    val logger = TestLogger()
    lateinit var communicator: SantaCommunicator

    beforeEach {
        communicator = SantaCommunicator(numberOfDaysToRest)
    }

    describe("composeMessage") {
        it("should compose correct message") {
            communicator
                .composeMessage(
                    "Dasher",
                    "North Pole",
                    5,
                    numberOfDayBeforeChristmas
                ) shouldBe "Dear Dasher, please return from North Pole in 17 day(s) to be ready and rest before Christmas."
        }
    }

    describe("isOverdue") {
        it("should detect overdue reindeer") {
            val overdue = communicator.isOverdue(
                "Dasher",
                "North Pole",
                numberOfDayBeforeChristmas,
                numberOfDayBeforeChristmas,
                logger
            )

            overdue shouldBe true
            logger.getLog() shouldBe "Overdue for Dasher located North Pole."
        }

        it("should return false when not overdue") {
            communicator.isOverdue(
                "Dasher",
                "North Pole",
                numberOfDayBeforeChristmas - numberOfDaysToRest - 1,
                numberOfDayBeforeChristmas,
                logger
            ) shouldBe false
        }
    }
})
